src/rrr/reactor/reactor.cc

```diff
@@ -677,55 +677,55 @@ static inline uint32_t hash_fd(uint32_t key) {
   return key;
 }
 
-void PollMgr::add(shared_ptr<Pollable> poll) {
+void PollMgr::add(shared_ptr<Pollable> poll) const {
```

Comment:
I am not sure why this can be made const, the add(poll) looks a non-const function?

Reply:
and similar for all/most these below

Reply:
I am not sure how these got past compiling, as it is calling a non-const function inside a const function

Reply:
I think that the these functions are changing just the poll_threads member variable which is not a member variable of the PollMgr class, which essentially modifies the member variables of the PollThread class and not the PollMgr class.

Reply:
But, in order to make it work, you have to change the functions in PollThread to const too, no?

Reply:
It's not necessary it seems... although I'm confused about this too

Reply:
Reason for this: https://stackoverflow.com/questions/7459867/preventing-a-const-member-function-from-changing-a-member-array

Apparently, we should never use array, just use vector.

Reply:
Oh ok, ...let me change this to vector and see what I can do about this.

Reply:
Some pretty confusing issues arise when I try to change the pollmgr to mut_ptr. I think this mainly arises because we are storing a reference inside the "Client" class for a pollmgr which requires us to pass around that pointer. This causes an issue because whenever the Client is destructed it also deletes the reference of that pollmgr which throws an error (...I think that is what happens...)

I tried fixing this by passing around the pollmgr through Client's functions but this causes an issue because ClientPool requires us to keep track of which Client is associated with which pointer to clean up in its destructor which becomes impossible here https://github.com/stonysystems/dslabs-cpp/blob/ae6291def6fdb6b51b779516d7e888aa93c5315a/src/rrr/rpc/client.cpp#L575 

I think using a shared_ptr over here makes sense as that will be simpler to use. WDYT